modules/indexing/src/test/java/org/apache/ignite/util/GridCommandHandlerMetadataTest.java

```diff
@@ -0,0 +1,312 @@
+     * - checks that type removed (try to create Type0 with the same field and different type).
+     * - removes the new type 'Type0' by cmd line utility.
+     * - restores Typo0 from the file
+     * - checks restored type.
+     */
+    @Test
+    public void testRemoveUpdate() throws Exception {
+        injectTestSystemOut();
+
+        Path typeFile = FS.getPath("type0.bin");
+
+        try {
+            createType("Type0", 0);
+
+            // Type of the field cannot be changed.
+            GridTestUtils.assertThrowsAnyCause(log, () -> {
+                createType("Type0", "string");
+
+                return null;
+            }, BinaryObjectException.class, "Wrong value has been set");
+
+            assertEquals(EXIT_CODE_OK, execute("--meta", "remove",
+                "--typeName", "Type0",
+                "--out", typeFile.toString()));
+
+            // New type must be created successfully after remove the type.
+            createType("Type0", "string");
+
+            assertEquals(EXIT_CODE_OK, execute("--meta", "remove", "--typeName", "Type0"));
+
+            // Restore the metadata from file.
+            assertEquals(EXIT_CODE_OK, execute("--meta", "update", "--in", typeFile.toString()));
+
+            // Type of the field cannot be changed.
+            GridTestUtils.assertThrowsAnyCause(log, () -> {
+                createType("Type0", "string");
+
+                return null;
+            }, BinaryObjectException.class, "Wrong value has been set");
+
+            createType("Type0", 1);
+
+            crd.context().cacheObjects().removeType(crd.context().cacheObjects().typeId("Type0"));
+
+            createType("Type0", "string");
+
+            // Restore the metadata from file.
+            assertEquals(EXIT_CODE_UNEXPECTED_ERROR, execute("--meta", "update", "--in", typeFile.toString()));
+
+            String out = testOut.toString();
+
+            assertContains(log, out, "Failed to execute metadata command='update'");
+            assertContains(log, out, "Type 'Type0' with typeId 110843958 has a " +
+                "different/incorrect type for field 'fld'.");
+            assertContains(log, out, "Expected 'String' but 'int' was provided. " +
+                "Field type's modification is unsupported");
+        }
+        finally {
+            if (Files.exists(typeFile))
+                Files.delete(typeFile);
+        }
+    }
+
+    /**
+     * Check the all thin connections are dropped on the command '--meta remove' and '--meta update'.
+     * Steps:
+     * - opens thin client connection.
+     * - creates Type0 on thin client side.
+     * - removes type by cmd line util.
+     * - executes any command on thin client to detect disconnect.
+     * - creates Type0 on thin client side with other type of field (checks the type was removed).
+     */
+    @Test
+    public void testDropThinConnectionsOnRemove() throws Exception {
```

Comment:
Let's add test like this but for JDBC thin client, to get sure that all works fine